backend-config/namespace_config.go

```diff
@@ -80,10 +86,18 @@ func (nc *namespaceConfig) getFromAPI(ctx context.Context) (map[string]ConfigT,
 	if nc.namespace == "" {
```

Comment:
We should be able to delete the line above, right? i.e. `config := make(map[string]ConfigT)`.

Reply:
It seems it won't work because the upstream code which uses this call actually ends up using the return value even if there is an error. It might require make some changes there as well if we were to change this

Reply:
That's unfortunate. I suggest we rename `config` to `configOnError` to at least make clear that the map isn't really meant for anything but it's just a "zero" value that we return in case of error.

Reply:
Done